src/ucp/core/ucp_context.c

```diff
@@ -851,6 +869,60 @@ static void ucp_fill_sockaddr_aux_tls_config(ucp_context_h context,
     }
 }
 
+static void ucp_fill_sockaddr_prio_list(ucp_context_h context,
+                                        const ucp_config_t *config)
+{
+    const char **sockaddr_tl_names = (const char**)config->sockaddr_cm_tls.cm_tls;
+    unsigned num_cfg_sockaddr_tls  = config->sockaddr_cm_tls.count;;
+    int                        idx = 0;
+    ucp_tl_resource_desc_t *resource;
+    ucp_rsc_index_t tl_id;
+    ucp_tl_md_t *tl_md;
+    int j;
+
+    /* Check if a list of sockadrr transports exists */
+    if (config->sockaddr_cm_tls.count == 0) {
+        ucs_debug("No sockaddr transports specified");
+        return;
+    }
+
+    /* Parse the sockadrr transports priority list */
+    for (j = 0; j < num_cfg_sockaddr_tls; j++) {
+        /* go over the proiroty list and find the transport's tl_id in the
+         * context's tl_bitmap. save the tl_id's for the client/server usage later */
+        if (strncmp(sockaddr_tl_names[j], "*", 1)) {
+            ucs_for_each_bit(tl_id, context->tl_bitmap) {
+                resource = &context->tl_rscs[tl_id];
+                tl_md    = &context->tl_mds[resource->md_index];
+                if (!strcmp(config->sockaddr_cm_tls.cm_tls[j],
+                            resource->tl_rsc.tl_name) &&
+                    (tl_md->attr.cap.flags & UCT_MD_FLAG_SOCKADDR) &&
+                    (ucp_find_rsc_array(context->config.sockaddr_tl_ids,
+                                        idx, tl_id) == -1)) {
+                    context->config.sockaddr_tl_ids[idx] = tl_id;
+                    idx++;
+                    break;
+                }
+            }
+        } else {
+            /* wildcard - search for all the available sockaddr transports
+             * and add them to the list */
+            ucs_for_each_bit(tl_id, context->tl_bitmap) {
```

Comment:
need to unite with the previous loop (line 894) to reduce code duplication

Reply:
but for *, need to start the search from the begining (of the context bitmap)

Reply:
but both loops already start from the beginning
BTW we can use just for (tl_id=0;tl_id<context->num_tls;++tl_id), no need for ucs_for_each_bit

Reply:
but they need to start over again on every entry to if/else.

why not use the bitmap? it has only the used tls.